src/autoscaler/api/brokerserver/broker_handler.go

```diff
@@ -56,17 +57,25 @@ func (h *BrokerHandler) CreateServiceInstance(w http.ResponseWriter, r *http.Req
 	instanceId := vars["instanceId"]
 
 	body := &models.InstanceCreationRequestBody{}
-	err := json.NewDecoder(r.Body).Decode(&body)
+	bodyBytes, err := ioutil.ReadAll(r.Body)
 	if err != nil {
-		h.logger.Error("failed to create service instance when trying to read request body", err)
+		h.logger.Error("failed to read service provision request body", err, lager.Data{"instanceId": instanceId})
+		handlers.WriteJSONResponse(w, http.StatusInternalServerError, models.ErrorResponse{
+			Code:    "Interal-Server-Error",
+			Message: "Failed to read request body"})
+		return
+	}
+	err = json.Unmarshal(bodyBytes, body)
+	if err != nil {
+		h.logger.Error("failed to unmarshal service provision body", err, lager.Data{"instanceId": instanceId, "body": string(bodyBytes)})
 		handlers.WriteJSONResponse(w, http.StatusBadRequest, models.ErrorResponse{
 			Code:    "Bad Request",
 			Message: "Invalid request body format"})
 		return
 	}
 
-	if instanceId == "" || body.OrgGUID == "" || body.SpaceGUID == "" || body.ServiceID == "" || body.PlanID == "" {
-		h.logger.Error("failed to create service instance when trying to get mandatory data", nil, lager.Data{"instanceId": instanceId, "orgGuid": body.OrgGUID, "spaceGuid": body.SpaceGUID, "serviceId": body.ServiceID, "planId": body.PlanID})
+	if instanceId == "" || body.OrgGUID == "" || body.SpaceGUID == "" {
+		h.logger.Error("failed to create service instance when trying to get mandatory data", nil, lager.Data{"instanceId": instanceId, "orgGuid": body.OrgGUID, "spaceGuid": body.SpaceGUID})
```

Comment:
according to https://github.com/openservicebrokerapi/servicebroker/blob/master/spec.md#provisioning,  serviceID and planID must be provided in request body